pkg/controller/dns/resource_record_set.go

```diff
@@ -0,0 +1,252 @@
+	cr, ok := mg.(*v1alpha1.ResourceRecordSet)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotResourceRecordSet)
+	}
+
+	rrs, err := e.Get(ctx, cr)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(
+			resource.Ignore(gcp.IsErrorNotFound, err),
+			errGetFailed,
+		)
+	}
+
+	lateInit := false
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	rrsClient.LateInitializeSpec(&cr.Spec.ForProvider, *rrs)
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errManagedUpdateFailed)
+		}
+		lateInit = true
+	}
+
+	cr.Status.AtProvider = rrsClient.GenerateObservation(*rrs)
+	cr.SetConditions(xpv1.Available())
+
+	upToDate, err := rrsClient.IsUpToDate(meta.GetExternalName(cr), &cr.Spec.ForProvider, rrs)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errCheckUpToDate)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:          true,
+		ResourceUpToDate:        upToDate,
+		ResourceLateInitialized: lateInit,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.ResourceRecordSet)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotResourceRecordSet)
+	}
+
+	args := &dns.ResourceRecordSet{}
+	rrsClient.GenerateResourceRecordSet(
+		meta.GetExternalName(cr),
+		cr.Spec.ForProvider,
+		args,
+	)
+
+	rrs, err := e.dns.Create(
+		e.projectID,
+		cr.Spec.ForProvider.ManagedZone,
+		args,
+	).Context(ctx).Do()
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCannotCreate)
+	}
+
+	cr.Status.AtProvider = rrsClient.GenerateObservation(*rrs)
+	cr.SetConditions(xpv1.Creating())
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.ResourceRecordSet)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotResourceRecordSet)
+	}
+
+	args := &dns.ResourceRecordSet{}
+	rrsClient.GenerateResourceRecordSet(meta.GetExternalName(cr), cr.Spec.ForProvider, args)
+
+	rrs, err := e.dns.Patch(
+		e.projectID,
+		cr.Spec.ForProvider.ManagedZone,
+		cr.Status.AtProvider.Name,
+		cr.Status.AtProvider.Type,
+		args,
+	).Context(ctx).Do()
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	cr.Status.AtProvider = rrsClient.GenerateObservation(*rrs)
+
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
+	cr, ok := mg.(*v1alpha1.ResourceRecordSet)
+	if !ok {
+		return errors.New(errNotResourceRecordSet)
+	}
+
+	_, err := e.dns.Delete(
+		e.projectID,
+		cr.Spec.ForProvider.ManagedZone,
+		meta.GetExternalName(cr),
+		cr.Spec.ForProvider.Type,
+	).Context(ctx).Do()
+	if gcp.IsErrorNotFound(err) {
+		return nil
+	}
+
+	cr.SetConditions(xpv1.Deleting())
```

Comment:
I think we can drop this line given the managed reconciler already handling this for us: https://github.com/crossplane/crossplane-runtime/blob/ba474e81c62c8e92a810d471a9a04efa62245817/pkg/reconciler/managed/reconciler.go#L636

Reply:
Oh, thanks. I didn't know that